lib/kitchen/element_base.rb

```diff
@@ -508,6 +508,22 @@ def parent
       Element.new(node: raw.parent, document: document, short_type: "parent(#{short_type})")
     end
 
+    # returns previous element
+    # skips double indentations that the nokigiri sometimes picks up
+    # nil if there's no previous sibling
+    #
+    def previous
+      node = raw.previous
+      return if node.nil?
+
+      sibling = Element.new(
+        node: node,
+        document: document,
+        short_type: "previous(#{short_type})"
+      )
+      sibling.to_s =~ /\S/ ? sibling : sibling.previous
```

Comment:
What is this line doing?  Looking backwards until it finds a non whitespace element sibling?

If so, probably more efficient to do that at the start of the method so we don't make a bunch of unused `Element` objects.  Actually can probably add to `lib/kitchen/patches/nokogiri.rb`:

```ruby
module Nokogiri
  class Node
    def previous_element
      prev = previous
      return nil if prev.nil?

      prev.text? ? prev.previous_element : prev
    end
  end
end
```

then change your first line above to 

```ruby
node = raw.previous_element
```

and ditch the last line

Reply:
Note this is also more efficient b/c it isn't serializing the element to string a lot